File: Top 100 Codes/05-Operations on Strings/01-Check whether a character is a vowel or consonant/Checkwhethercharacterisavowelorconsonant.cpp
#include <iostream>
using namespace std;

// Function to check if the character is a vowel or consonant
void checkCharacter(char ch) {
    // Convert the character to lowercase for uniform comparison
    ch = tolower(ch);

    if (ch == 'a' || ch == 'e' || ch == 'i' || ch == 'o' || ch == 'u') {
        cout << ch << " is a vowel." << endl;
    } 
    else if (ch >= 'a' && ch <= 'z') {
        cout << ch << " is a consonant." << endl;
    } 
    else {
        cout << "Invalid input. Please enter an alphabetic character." << endl;
    }
}

int main() {
    char ch;
    cout << "Enter a character: ";
    cin >> ch;

    // Call function to check character type
    checkCharacter(ch);

    return 0;
}

File: Top 100 Codes/05-Operations on Strings/01-Check whether a character is a vowel or consonant/Checkwhethercharacterisavowelorconsonant.py
def check_character(ch):
    # Convert the character to lowercase for uniform comparison
    ch = ch.lower()

    if ch in ('a', 'e', 'i', 'o', 'u'):
        print(f"{ch} is a vowel.")
    elif 'a' <= ch <= 'z':  # Check if it's an alphabet character
        print(f"{ch} is a consonant.")
    else:
        print("Invalid input. Please enter an alphabetic character.")

# Taking user input
ch = input("Enter a character: ")

# Ensuring input is a single character
if len(ch) == 1:
    check_character(ch)
else:
    print("Please enter a single character.")
